Helper/Cart.php

```diff
@@ -399,7 +460,8 @@ public function getCartData($payment_only, $place_order_payload)
                 'country_code'    => $shippingAddress->getCountryId(),
             ];
 
-            if (@$email) {
+            $email = $shippingAddress->getEmail() ?: $email;
+            if ($email) {
```

Comment:
In case of payment only, current code overrides shipping email with billing email even if user provides different emails in magento checkout page.

Reply:
There is only one place to enter email in magento checkout. It is at some point added to shipping and billing addresses. If the shipping email exists that one will be used, otherwise the email that user entered is collected from the page and sent with the request ($place_order_payload) and that one will be used. If that can not be found and there is an email stored with billing, that one is used. But in practice it is all one email.